src/sphinxcontrib/towncrier/_fragment_discovery.py

```diff
@@ -42,11 +47,21 @@ def _find_config_file(base: Path) -> Path:
 # pylint: disable=fixme
 # FIXME: refactor `lookup_towncrier_fragments` to drop noqas
 @lru_cache(maxsize=1, typed=True)  # noqa: WPS210
-def lookup_towncrier_fragments(  # noqa: WPS210
+def lookup_towncrier_fragments(
         working_dir: Optional[str] = None,
         config_path: Optional[str] = None,
 ) -> Set[Path]:
     """Emit RST-formatted Towncrier changelog fragment paths."""
+    tc_version = Version(version('towncrier'))
+    if tc_version >= Version('24.7.0'):
+        return _lookup_towncrier_fragments_post24_7(working_dir, config_path)
+    return _lookup_towncrier_fragments_pre24_7(working_dir, config_path)
+
+
+# used for towncrier version 24.7 and above
+def _lookup_towncrier_fragments_post24_7(  # noqa: WPS210
```

Comment:
(this needs to be communicated *somehow*)
```suggestion
def _lookup_towncrier_fragments_post24_7_inclusive(  # noqa: WPS210
```